### ISSUE TITLE
Reproducing results of the paper "Estimating and Evaluating the Uncertainty of Rating Predictions and Top-n Recommendations in Recommender Systems" (Using the code of SnapExplicit)

### ISSUE THREAD

**Issue by coralscharf** [vcoscrato/uncertain#1]
In the train function, the monitor of EarlyStopping and ModelCheckpoint is set to val_MAP. However, you didn't log this value during training. Moreover, in the paper, you wrote that the model hyperparameters values are the values that give the lowest validation set RMSE but it seems that the selection is done based on the validation set loss. Can you please provide information regarding your training procedure?

**Comment by vcoscrato** [comment]
Thanks for raising this issue.

First, we noticed that there was a versioning error on the file uncertain/utils/training.py. We made available a version that is based on some later work.. The correct version for this snapshot branch, corresponding to the paper, is now available. Notice that, on the correct version, we log ‘val_loss’, referring to loss on the validation set.

Second, we acknowledge the confusion between RMSE and validation loss. In the paper, we simplify by writing that the tuning is always performed according to RMSE. In fact, it is done using validation loss; and some of the models (e.g. CPMF and OrdRec) do not use RMSE as their validation loss. Therefore, arguably the paper should say that the tuning and the early stopping choose the model with the lowest validation loss.

To reproduce the results from the paper, please follow the notebook at tests/Movielens.ipynb and tests/Netflix.ipynb. There, you will find the full training procedure. Please, let us know about any further concerns regarding this repo.
